day1/part2.py

```diff
@@ -1,43 +1,31 @@
-with open('input.txt', 'r') as f:
-    numbers = ('one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine')
-    lines = list(map(str.rstrip, f.readlines()))
-    num_of_line = 0
-    counter = 0
-    for line in lines:
-        counter += 1
-        for character in line:
-            if character.isdigit():
-                first = {character: line.index(character)}
-                break
-        for number in numbers:
-            if line.find(number) != -1:
-                first.update({number: line.index(number)})
-
-        for character in line[::-1]:
+def find_digit(ln: str, directio: int, numbers, my_dist):
```

Comment:
The same as above/ Remove the direction parameter. Also better naming practice would be to name the whole direction.